api/serviceregistry/serviceregistry.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+
+package svreg;
+option go_package = "serviceregistry";
+
+
+service Registry{
+  rpc RegisterWorkload (ServiceWorkload) returns (Empty) {} // Request register of a new service
+  rpc QueryService (ServiceRequest) returns (ServiceWorkloads) {} // Request a services
+  rpc RemoveWorkload (ServiceWorkload) returns (Empty) {} // Remove service from list
+}
+
+message Empty {}
+
+//The request used to register a new service or update the existing ones
+message ServiceWorkload {
+  string serviceName = 1;
+  string connectivityDomain = 2;
+  Workload workload = 3;
+  repeated int32 ports = 4;
+}
+
+//Basic workload
```

Comment:
Style nit: Please try to keep items that work together next to each other in the proto definition. The Workload item is here, in between 2 Service type items, and the WorkloadIdentifier, which is only used by the workload is at the bottom.

Reply:
done